tests/func/test_catalog.py

```diff
@@ -840,13 +840,17 @@ def test_listing_stats(cloud_test_catalog):
 
     catalog.enlist_source(f"{src_uri}/dogs/", update=True)
     stats = listing_stats(src_uri, catalog)
+    assert stats.num_objects == 7
```

Comment:
[C]: not sure why we expected here any change to the `src_uri` listing, after listing the subset. It seems it is better to create a second listing for `/dogs` in this case (or at least update the whole `src_uri`).

Reply:
cc @ilongin you might know this better

Reply:
I think you are right and this was a mistake / bug in the listing. Creating new separate `/dogs` seems like correct thing to do here.